### PR TITLE
RHOAIENG-10827: feat(nbcs): update ose-oauth-proxy image digest reference from 4.8 to the latest 4.14 version

### PR DESCRIPTION
**Pull request by jiridanek**
## Description

https://issues.redhat.com/browse/RHOAIENG-10827

## How Has This Been Tested?

Wait for OCP-CI to build this and deploy it in some RHOAI.

* quay.io/opendatahub/kubeflow-notebook-controller:pr-386
* quay.io/opendatahub/odh-notebook-controller:pr-386

Need to also edit args in the odh-notebook-controller deployment because normally the oauth proxy digest arg comes from rhods operator.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by shalberd** [comment]
yeah, that is what I am currently using as well on OCP 4.14.
However, pre-4.14 ose-oauth images did not pose any problems, I just made it good practice to override the default in the manifests. Good that default in code is now 4.14.

**Comment by shalberd** [comment]
Does it make sense to leave imagePullPolicy at Always in

https://github.com/opendatahub-io/kubeflow/blob/v1.7-branch/components/odh-notebook-controller/controllers/notebook_webhook.go#L78

vs

corev1.PullIfNotPresent?

File: components/odh-notebook-controller/config/manager/manager.yaml
Comment (jstourac) [inline components/odh-notebook-controller/config/manager/manager.yaml:28]: Hmmm, how did you get this sha? :thinking: When I check the PR for the Dashboard opendatahub-io/odh-dashboard#3216, it uses the following sha:
```
4f8d66597feeb32bb18699326029f9a71a5aca4a57679d636b876377c2e95695
```

Also, when I tried to create an ImageStream on my OpenShift cluster, I can see following:
```
kind: ImageStreamTag
apiVersion: image.openshift.io/v1
metadata:
  name: 'example:v4.14'
  namespace: default
  uid: f41c7b88-c7e3-4e56-bc11-bc389053e033
  resourceVersion: '44113055'
  creationTimestamp: '2024-09-19T07:45:52Z'
tag:
  name: v4.14
  annotations: null
  from:
    kind: DockerImage
    name: 'registry.redhat.io/openshift4/ose-oauth-proxy:v4.14'
  generation: 2
  importPolicy:
    scheduled: true
    importMode: PreserveOriginal
  referencePolicy:
    type: Source
generation: 2
lookupPolicy:
  local: false
image:
  metadata:
    name: 'sha256:4f8d66597feeb32bb18699326029f9a71a5aca4a57679d636b876377c2e95695'
    uid: bcfc4de2-b22b-49fb-9d1e-c2d65ddb4201
    resourceVersion: '44113054'
    creationTimestamp: '2024-09-19T07:45:52Z'
  dockerImageReference: 'registry.redhat.io/openshift4/ose-oauth-proxy@sha256:4f8d66597feeb32bb18699326029f9a71a5aca4a57679d636b876377c2e95695'
  dockerImageMetadata:
    kind: DockerImage
    apiVersion: image.openshift.io/1.0
    Id: 'sha256:4f8d66597feeb32bb18699326029f9a71a5aca4a57679d636b876377c2e95695'
    Created: '2024-09-19T07:45:50Z'
    ContainerConfig: {}
  dockerImageMetadataVersion: '1.0'
  dockerImageManifestMediaType: application/vnd.docker.distribution.manifest.list.v2+json
  dockerImageManifests:
    - digest: 'sha256:105307b602ac74649f868e1ea0aab7b8621ea1ecfd58ceca669dcf32f538798e'
      mediaType: application/vnd.docker.distribution.manifest.v2+json
      manifestSize: 760
      architecture: amd64
      os: linux
    - digest: 'sha256:c55ea5f29f59eccbb57d5606d84ab53d9987b0226e1e93d76aefa22bc2b82d0b'
      mediaType: application/vnd.docker.distribution.manifest.v2+json
      manifestSize: 760
      architecture: arm64
      os: linux
    - digest: 'sha256:0eb617ef510f990b11a3144077bda3bc96bdd2a9fd19bff9d1f3d11cab243c79'
      mediaType: application/vnd.docker.distribution.manifest.v2+json
      manifestSize: 760
      architecture: ppc64le
      os: linux
    - digest: 'sha256:1cff8c85d0befc48805d4339913fd0e7f02896c1c0ad313e3e4ec2e001ccecb9'
      mediaType: application/vnd.docker.distribution.manifest.v2+json
      manifestSize: 760
      architecture: s390x
      os: linux
```

My guess is you used some manifests sha instead? Or am I missing something? :thinking:

Comment (jiridanek) [reply components/odh-notebook-controller/config/manager/manager.yaml:28]: I used image sha, what dashboard pr uses is manifests sha.

Comment (jiridanek) [reply components/odh-notebook-controller/config/manager/manager.yaml:28]: https://catalog.redhat.com/software/containers/openshift4/ose-oauth-proxy/5cdb2133bed8bd5717d5ae64?image=66cefc14401df6ff4664ec43&architecture=amd64&container-tabs=gti

Comment (atheo89) [reply components/odh-notebook-controller/config/manager/manager.yaml:28]: Thanks Jiri, eventually they changed it to platform agnostic https://github.com/opendatahub-io/odh-dashboard/pull/3216/commits/6d95c02312b6eeed25d3eaffd8cff5a976e829aa and use back the manifest digest

Comment (shalberd) [reply components/odh-notebook-controller/config/manager/manager.yaml:28]: correct. For a given tag, like v4.14, the digest changes every now and then, I think there is a stream kind of regular build process behind it.
Screenshot 2024-09-19 at 11 07 22
Screenshot 2024-09-19 at 11 07 34

Currently, the manifest list digest from the build 8 days ago is
`sha256:4f8d66597feeb32bb18699326029f9a71a5aca4a57679d636b876377c2e95695`

**Review by jstourac — APPROVED** [review]
/lgtm

**Comment by jstourac** [comment]
Note: @jiridanek once this is merged, we shall cherry-pick this to stable branch (as you already tried), but eventually also into the 1.9 branch.

**Comment by jiridanek** [comment]
Works for me, the oauth image changed from registry.redhat.io/openshift4/ose-oauth-proxy@sha256:4bef31eb993feb6f1096b51b4876c65a6fb1f4401fee97fa4f4542b6b7c9bc46 to registry.redhat.io/openshift4/ose-oauth-proxy@sha256:4f8d66597feeb32bb18699326029f9a71a5aca4a57679d636b876377c2e95695 and I was able to auth into my workbench. Imo this is good enough to go into odh-nightly for more testing, but I'd leave that decision on the others. I will put that on hold, pending grooming and planning on the linked Jira issue.

**Pull request merged**
